tests/unit/lib/Any_unittest.cc

```diff
@@ -636,3 +641,40 @@ TEST(Any, reset_array2d_reference)
 	EXPECT_EQ(obj->ref_count(), 1);
 	SG_UNREF(obj);
 }
+
+TEST(Any, lazy_simple)
+{
+	auto v = 9;
+	auto any = make_any<int>([=]() { return v; });
+	EXPECT_EQ(any.as<int32_t>(), v);
+}
+
+TEST(Any, lazy_assignment_into_with_value)
+{
+	auto v = 3;
+	auto any = make_any<int>([=]() { return 111; });
+	EXPECT_THROW(any = make_any<int>(v), TypeMismatchException);
+}
+
+TEST(Any, lazy_assignment_into_with_function)
+{
+	auto v = 3;
+	auto any = make_any<int>([=]() { return 111; });
+	any = make_any<int>([=]() { return v; });
+	EXPECT_EQ(any.as<int32_t>(), v);
+}
+
+TEST(Any, lazy_assignment_from)
+{
+	auto v = 3;
+	auto any = make_any(0);
+	EXPECT_THROW(
+	    any = make_any<int>([=]() { return v; }), TypeMismatchException);
+}
+
+TEST(Any, lazy_member_function)
```

Comment:
What do you think about a test case that asserts laziness?

Reply:
Good idea

Reply:
Also, we need to keep in mind equals/serialization should exclude lazy computations.

Reply:
i.e. we need some kind of markup. Otherwise these things will be equaled/cloned/serialized as well....or is that intended? Costs computation and I am not sure it is worth it

Reply:
ah, clone_from will be a problem as it attempts to write into the member

Reply:
That's a bunch of questions :)

- Serialization is tricky. It seems the 'serializator' itself should know
- We can explicitly throw something when trying to do equals
- Clone is cheap but it won't work correctly. It would use the original object

Reply:
I think then in clone and equals, I would just skip all registered function anys